packages/scaffold/src/partials/w3m-header/index.ts

```diff
@@ -116,7 +117,7 @@ export class W3mHeader extends LitElement {
   }
 
   private async onClose() {
-    if (OptionsController.state.isSiweEnabled) {
+    if (OptionsController.state.isSiweEnabled && SIWEController.state.status === 'loading') {
```

Comment:
SIWEController.state.status !== 'success'
```suggestion
    if (OptionsController.state.isSiweEnabled && SIWEController.state.status !== 'success') {
```